components/tabs/src/sync/store.rs

```diff
@@ -103,7 +116,7 @@ impl TabsStore {
             return Err(e.into());
         }
         match result.engine_results.remove("tabs") {
-            None | Some(Ok(())) => Ok(result.telemetry),
+            None | Some(Ok(())) => Ok(serde_json::to_string(&result.telemetry).unwrap()),
```

Comment:
We expose an error for serde errors, so I think this `.unwrap()` can be replaced with `?`